backend/util/helpers.js

```diff
@@ -0,0 +1,206 @@
+const { faker } = require('@faker-js/faker');
+const { Product } = require('../models/product');
+
+async function addFakeData (req, res, next) {
+  for (let i = 0; i < 100; i++) {
+    let product = new Product();
+    const numReviews = Math.floor(Math.random() * 8);
+
+    product.category = faker.commerce.department();
+    product.name = faker.commerce.productName();
+    product.price = faker.commerce.price();
+    product.image = 'https://via.placeholder.com/250?text=Product+Image';
+    
+    // Create fake reviews
+    for (let j = 0; j < numReviews; j++) {
+      const username = faker.internet.userName();
+      const text = faker.lorem.sentence();
+      
+      product.reviews.push(
+        {
+          'username': username,
+          'text': text,
+        }
+      );
+    }
+
+    await product.save()
+      .catch(err => console.error(err));
+  }
+  res.status(200).send('Products saved');
+}
+
+async function getProducts(req, res, next) {
+  let page = Number(req.query.page); 
+  const { category, price, query } = req.query;
+  
+  // Check that the value of page is correct and default to 1 if not
+  if (page <= 0 || isNaN(page)) { page = 1 }
+  
+  const limit = 9; // Limits the returned products to 9 per page
+  const start = (page - 1) * limit;
+  const stop = page * limit;
+  const queryFilter = {};
+
+  // If a category was included, add it to the query object.
+  if (!!category) { queryFilter.category = category };
+
+  // If a query parameter for 'query' was included, add it to the query object.
+  if (!!query) { 
+    queryFilter.$or = [
+      { name: { $regex: query, $options: 'i' } }, 
+      { category: { $regex: query, $options: 'i' } }
+    ]; 
+  }
+
+  // Stores all results returned by MongoDB
+  const results = await Product.find(queryFilter);
+  const resultsCount = results.length;
+  const totalPages = Math.ceil(resultsCount / limit);
+
+  // Sort the results if the price query value is included
+  if (price === 'lowest') {
+    results.sort((a, b) => a.price - b.price);
+  }
+
+  if (price === 'highest') {
+    results.sort((a, b) => b.price - a.price);
+  }
+
+  // Slices the array to return only 9 results per page
+  const resultsLimited = results.slice(start, stop);
+
+  // Sends the paginated results to the client
+  res.send({ resultsCount, totalPages, page, resultsLimited, query, category });
+}
+
+// A callback function to pull all categories
+async function getCategories(req, res, next) {
+  // Query all products from MongoDB
+  const results = await Product.find({});
```

Comment:
be more specific, results of what?
```suggestion
  const allProducts = await Product.find({});
```